### ISSUE TITLE
Using -o yaml and --query returns strings followed by ellipses

### ISSUE THREAD

**Issue by varndorfer** [Azure/azure-cli#25238]
**Related command**
az vm show -n test -o yaml --query name

**Describe the bug**
For any entity, using the show command with the `-o yaml` and `--query` flags together, the CLI returns all string values followed by "...". Dictionary values appear fine, for example:
<img width="454" alt="image" src="https://user-images.githubusercontent.com/20168613/214736909-31f0a066-581f-45e3-802c-4819578d4e0a.png">

**To Reproduce**
Works for any entity, but using AzureML job as an example. Find an existing job and use the show command with the `-o yaml` and `--query name` flags. e.g. `az ml job show -n "job_name" -o yaml --query name`

**Expected behavior**
The output should only have the queried value. No ellipses.

**Environment summary**

Tested using az version 2.30.0 with powershell and bash

**Comment by yonzhan** [comment]
@zhoxing-ms for awareness

**Comment by zhoxing-ms** [comment]
> Related command
az vm show -n test -o yaml --query name

I believe that the `Related command` written here is not correct, because the screenshot shows the `az ml job show` command. Since the `ml` extension is owned by service team, add the service team members @demoray to help take a look~

**Comment by jiasli** [comment]
## TL;DR answer

Don't use `--output yaml` when you want to extract the raw value of a property.

In Azure CLI, the correct way to exact the raw value is to use `--output tsv`. See

- https://learn.microsoft.com/en-us/cli/azure/format-output-azure-cli#tsv-output-format
- https://learn.microsoft.com/en-us/cli/azure/use-cli-effectively?tabs=bash%2Cbash2#output-formatting

## Full explanation

`...` is the boundary marker in YAML: https://yaml.org/spec/1.1/#id897596

> a YAML [document](https://yaml.org/spec/1.1/#document/syntax) may be terminated by an explicit end line denoted by “...”

This is the expected behavior of YAML if you want to dump scalar at the root level:

https://stackoverflow.com/questions/56950391/yaml-end-always-dumped-even-if-yaml-explicit-end-false

Below is the implementation detail of `pyyaml`.

Consider

```py
import yaml

print(yaml.safe_dump("hello world", explicit_end=True))
print(yaml.safe_dump("hello world", explicit_end=False))
```

There are 2 places where `...` is written when document ends:

https://github.com/yaml/pyyaml/blob/957ae4d495cf8fcb5475c6c2f1bce801096b68a5/lib/yaml/emitter.py#L206
https://github.com/yaml/pyyaml/blob/957ae4d495cf8fcb5475c6c2f1bce801096b68a5/lib/yaml/emitter.py#L218

When dumping a string and `explicit_end` is set to `True`, L218 from `expect_document_end` is called:

https://github.com/yaml/pyyaml/blob/957ae4d495cf8fcb5475c6c2f1bce801096b68a5/lib/yaml/emitter.py#L218

```py
    def expect_document_end(self):
        ...
                self.write_indicator('...', True)
```

`write_indicator` marks `open_ended` as `False`:

https://github.com/yaml/pyyaml/blob/957ae4d495cf8fcb5475c6c2f1bce801096b68a5/lib/yaml/emitter.py#L809

```py
    def write_indicator(self, indicator, need_whitespace,
        ...
        self.open_ended = False
```

So L206 is no longer called.

https://github.com/yaml/pyyaml/blob/957ae4d495cf8fcb5475c6c2f1bce801096b68a5/lib/yaml/emitter.py#L206

```py
            if self.open_ended:
                self.write_indicator('...', True)
```

On contrary, when dumping a string and `explicit_end` is set to `False`, L218 is skipped and `open_ended` is now `True`. Later on, L206 is called:

https://github.com/yaml/pyyaml/blob/957ae4d495cf8fcb5475c6c2f1bce801096b68a5/lib/yaml/emitter.py#L206

```py
            if self.open_ended:
                self.write_indicator('...', True)
```

This makes `...` still be written.
